core/src/main/java/cucumber/runtime/Runtime.java

```diff
@@ -190,7 +192,14 @@ public Runtime build() {
                 plugins.addPlugin(plugin);
             }
 
-            final RunnerSupplier runnerSupplier = new ThreadLocalRunnerSupplier(this.runtimeOptions, this.eventBus, backendSupplier, this.glueSupplier);
+            final RunnerSupplier runnerSupplier = runtimeOptions.isMultiThreaded()
+                ? new ThreadLocalRunnerSupplier(this.runtimeOptions, this.eventBus, backendSupplier, this.glueSupplier)
+                : new SingletonRunnerSupplier(this.runtimeOptions, this.eventBus, backendSupplier, this.glueSupplier);
+
+            final ExecutorService executor = runtimeOptions.isMultiThreaded()
```

Comment:
Given how often I am debugging tests it is not something to be sneezed at.

But it also makes debuging plugins easier because the events are provided as they are created rather then batched up.

Reply:
I wasn’t scoffing at the change at all. Just wonder if I’d missed something other than aiding in debugging tests - as I myself temporarily added it but didn’t commit it